prov/rxm/src/rxm_conn.c

```diff
@@ -1244,11 +1245,29 @@ rxm_conn_handle_reject(struct rxm_ep *rxm_ep, struct rxm_msg_eq_entry *entry)
 	return 0;
 }
 
+static int
+rxm_conn_handle_reset(struct rxm_ep *rxm_ep, struct rxm_msg_eq_entry *entry)
+{
+	struct rxm_cmap_handle *handle = entry->err_entry.fid->context;
+
+	if (handle->state == RXM_CMAP_CONNREQ_SENT) {
```

Comment:
ECONNRESET can occur on a connection at any time.  Is there a reason to check the state here, versus always just closing the handle?

Reply:
connection reset is expected and normal just after connect() call, and normal processing is to allow connect again, this is why I set state to 'idle'. but I'm not sure we should allow connect again if connection resets later (probably setting state to 'shutdown' is ok', but some testing required)

Reply:
If the peer application exits and restarts, a new connection is needed.  So, a connection reset can occur at any time, and the code should handle this.  We can't assume that the application is a tightly coupled parallel MPI job.  The patch only seems to deal with reset in a very narrow window.